lib/rucio/core/rse_selector.py

```diff
@@ -71,30 +72,54 @@ def __init__(self, account, rses, weight, copies, ignore_account_limit=False, se
                 rse['quota_left'] = float('inf')
                 rse['space_left'] = float('inf')
+                rses_with_enough_quota.append(rse)
         else:
+            global_quota_limit = get_global_account_limits(account=account, session=session)
+            all_rse_usages = {usage['rse_id']: usage['bytes'] for usage in get_all_rse_usages_per_account(account=account, session=session)}
             for rse in self.rses:
                 if rse['mock_rse']:
                     rse['quota_left'] = float('inf')
                     rse['space_left'] = float('inf')
+                    rses_with_enough_quota.append(rse)
                 else:
-                    quota_limit = get_account_limit(account=account, rse_id=rse['rse_id'], session=session)
+                    # check local quota
+                    local_quota_left = None
+                    quota_limit = get_local_account_limit(account=account, rse_id=rse['rse_id'], session=session)
                     if quota_limit is None:
-                        rse['quota_left'] = 0
+                        local_quota_left = 0
                     else:
-                        rse['quota_left'] = quota_limit - get_usage(rse_id=rse['rse_id'], account=account, session=session)['bytes']
-
-                    space_limit = get_rse_limits(name='MaxSpaceAvailable', rse_id=rse['rse_id'], session=session).get('MaxSpaceAvailable')
-                    if space_limit is None or space_limit < 0:
-                        rse['space_left'] = float('inf')
-                    else:
-                        rse['space_left'] = space_limit - get_rse_counter(rse_id=rse['rse_id'], session=session)['bytes']
-
-        self.rses = [rse for rse in self.rses if rse['quota_left'] > 0]
-
+                        local_quota_left = quota_limit - get_usage(rse_id=rse['rse_id'], account=account, session=session)['bytes']
+                    # check global quota
+                    all_global_quota_enough = True
+                    for rse_expression, limit in global_quota_limit.items():
+                        if rse['rse_id'] in limit['resolved_rse_ids']:
+                            quota_limit = limit['limit']
+                            global_quota_left = None
+                            if quota_limit is None:
+                                global_quota_left = 0
+                            else:
+                                rse_expression_usage = 0
+                                for rse_id in limit['resolved_rse_ids']:
+                                    rse_expression_usage += all_rse_usages.get(rse_id, 0)
+                                global_quota_left = quota_limit - rse_expression_usage
+                            if global_quota_left <= 0:
+                                all_global_quota_enough = False
+                                break
+                    if local_quota_left > 0 and all_global_quota_enough:
+                        rse['quota_left'] = local_quota_left
```

Comment:
Maybe I am not following the above code, but I think this is not correct.
This does not reflect the case where the global_quota is smaller than the local quota.
E.g. 100TB on the specific RSE, but only 1 TB left on the global one. I think the global quota structure needs to be persisted in the object, so it can later on also be updated in the `__update_quota(...)` call